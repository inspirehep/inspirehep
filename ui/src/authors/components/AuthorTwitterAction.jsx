import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'antd';

import ListItemAction from '../../common/components/ListItemAction';
import ExternalLink from '../../common/components/ExternalLink';

class AuthorTwitterAction extends Component {
  render() {
    const { twitter } = this.props;
    const href = `//twitter.com/${twitter}`;
    return (
      <ListItemAction>
        <ExternalLink href={href}>
          <Icon type="twitter" />
        </ExternalLink>
      </ListItemAction>
    );
  }
}

AuthorTwitterAction.propTypes = {
  twitter: PropTypes.string.isRequired,
};

export default AuthorTwitterAction;
