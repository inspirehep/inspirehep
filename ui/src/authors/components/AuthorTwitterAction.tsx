import React from 'react';
import { TwitterOutlined } from '@ant-design/icons';
import { Tooltip } from 'antd';

import UserAction from '../../common/components/UserAction';
import LinkWithTargetBlank from '../../common/components/LinkWithTargetBlank';
import EventTracker from '../../common/components/EventTracker';

const AuthorTwitterAction = ({ twitter }: { twitter: string }) => {
  const href = `//twitter.com/${twitter}`;
  return (
    <UserAction>
      <Tooltip title="Twitter">
        <EventTracker
          eventCategory="Author detail"
          eventAction="Link"
          eventId="Twitter"
        >
          <LinkWithTargetBlank href={href}>
            <TwitterOutlined />
          </LinkWithTargetBlank>
        </EventTracker>
      </Tooltip>
    </UserAction>
  );
};

export default AuthorTwitterAction;
