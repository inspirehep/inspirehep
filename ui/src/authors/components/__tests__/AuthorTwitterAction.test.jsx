import React from 'react';
import { shallow } from 'enzyme';

import AuthorTwitterAction from '../AuthorTwitterAction';

describe('AuthorTwitterAction', () => {
  it('renders with twitter', () => {
    const wrapper = shallow(<AuthorTwitterAction twitter="harunurhan" />);
    expect(wrapper).toMatchSnapshot();
  });
});
