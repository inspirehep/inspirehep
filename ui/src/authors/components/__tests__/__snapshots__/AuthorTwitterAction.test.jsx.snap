// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorTwitterAction renders with twitter 1`] = `
<ListItemAction>
  <ExternalLink
    href="//twitter.com/harunurhan"
  >
    <Icon
      type="twitter"
    />
  </ExternalLink>
</ListItemAction>
`;
